### PR TITLE
Update ValueTuple from review of public API

### PR DESCRIPTION
**Pull request by jcouv**
Two changes:
- The attribute uses "Item" as terminology instead of "Element"
- Made `ValueTuple'0` be a static class. It only serves as a factory now.

@weshaggard @stephentoub for review. Thanks

### CODE REVIEW

File: src/System.ValueTuple/tests/TupleElementNamesAttribute/UnitTests.cs [hunk 1]
```diff
@@ -1,43 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jcouv) [inline src/System.ValueTuple/tests/TupleElementNamesAttribute/UnitTests.cs]: This file was renamed and also had "Item" substituted for "Element".

**Comment by jcouv** [comment]
FYI for @gafter @VSadov

**Comment by jcouv** [comment]
test  Innerloop Windows_NT Release Build and Test please

**Comment by gafter** [comment]
I want to push back on this. The LDM’s current position is that there exists a 0-element tuple type. Even if we do not have syntactic support for it at this time, it is important for this to exist as an empty struct type today so we do not block ourselves from regularizing small (0-element and 1-element) tuples in the future.

**Comment by gafter** [comment]
Also, after LDM discussion, the parts of a tuple are likely to be termed *elements*, not items. So I want to push back on this aspect of the PR as well.

**Comment by jcouv** [comment]
Thanks @gafter for the clarifications.
FYI for @KrzysztofCwalina and @terrajobst. I will withdraw the PR and keep current public API for ValueTuples.

**Pull request closed without merging**